src/components/TodoList/TodoList.tsx

```diff
@@ -0,0 +1,36 @@
+import React, { useMemo } from 'react';
+import { Todo } from '../../types/Todo';
+import { TodoItem } from '../TodoItem';
+
+type Props = {
+  visibleToDos: Todo[];
+  onDelete: (id: number) => Promise<void>;
+  onUpdate: (id: number, updated: Partial<Todo>) => void;
+  tempTodo: Todo | null;
+};
+
+export const TodoList: React.FC<Props> = ({
+  visibleToDos,
+  onDelete,
+  onUpdate,
+  tempTodo,
+}) => {
+  const renderToDos = useMemo(
+    () => (tempTodo ? [...visibleToDos, tempTodo] : visibleToDos),
```

Comment:
Instead, you could add a conditional rendering for a <TodoItem> after the .map method.
In that way, you don't have to re-render the entire list because the tempTodo has changed.